sdk/core/azure-core-management/src/main/java/com/azure/core/management/profile/AzureProfile.java

```diff
@@ -26,7 +72,10 @@ public final class AzureProfile {
      * </ul>
      *
      * @param environment the Azure environment
+     * @deprecated use {@link AzureProfile#AzureProfile(AzureCloud)}
+     * @see AzureProfile#AzureProfile(AzureCloud)
      */
+    @Deprecated
```

Comment:
Maybe we should not deprecate it right now, till azure-identity released with `AzureCloud` support, and we have our documents updated.

Reply:
Yeah, I agree. 
Fact is, we probably don't want to deprecate this any time soon. If user wants to set custom endpoints, this is the current only way.
I'm facing kinda of a dilemma:
1. We want `AzureProfile(AzureCloud)` to reflect the consistent `AZURE_PUBLIC_CLOUD`
2. We want `AzureProfile(AzureEnvironment)` in order for user to set custom endpoints(Azure Spring is [doing this](https://github.com/Azure/azure-sdk-for-java/blob/00fccdee1f8b637e773253aa1ce3566e0188a8ef/sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/provider/AzureProfileOptionsProvider.java#L225))
3. Coexistence of these two constructors may confuse user

To resolve this, we could do:
1. In our README.MD, use the new `AzureProfile(AzureCloud)`
2. In `AzureProfile(AzureEnvironment)` javadoc, explicitly say that only use this constructor if you want to use custom endpoints, and link to the other ctor for normal use case.

Should be OK in my opinion.

Reply:
Ah, yes, for other cloud they still need to use the `AzureEnvironment`.

Reply:
removed deprecation and added Note to `AzureProfile(AzureEnvironment)` javadoc